build-tools/xaprepare/xaprepare/ToolRunners/GitRunner.cs

```diff
@@ -95,7 +95,16 @@ public string GetTopCommitHash (string workingDirectory = null, bool shortHash =
 			string runnerWorkingDirectory = DetermineRunnerWorkingDirectory (workingDirectory);
 
 			var runner = CreateGitRunner (runnerWorkingDirectory);
-			runner.AddArgument ("rev-parse");
+
+			// Get commit message of merge commit being built, to parse the HEAD of the source branch.
+			if (Context.BuildingAZPPullRequest) {
```

Comment:
This is where the additional Azure check mentioned above would be useful. Here, we assume that if we're on Azure, it's a merge commit which will most likely be false on developer's machine (and Jenkins, too?)